mmyolo/models/dense_heads/yolov7_head.py

```diff
@@ -1,84 +1,101 @@
-        prior_generator(dict): Points generator feature maps
-            in 2D points-based detectors.
-        loss_cls (:obj:`ConfigDict` or dict): Config of classification loss.
-        loss_bbox (:obj:`ConfigDict` or dict): Config of localization loss.
-        loss_obj (:obj:`ConfigDict` or dict): Config of objectness loss.
-        train_cfg (:obj:`ConfigDict` or dict, optional): Training config of
-            anchor head. Defaults to None.
-        test_cfg (:obj:`ConfigDict` or dict, optional): Testing config of
-            anchor head. Defaults to None.
-        init_cfg (:obj:`ConfigDict` or list[:obj:`ConfigDict`] or dict or
-            list[dict], optional): Initialization config dict.
-            Defaults to None.
-    """
+    """YOLOv7Head head used in `YOLOv7 <https://arxiv.org/abs/2207.02696>`_."""
 
     def __init__(self,
-                 head_module: nn.Module,
-                 prior_generator: ConfigType = dict(
-                     type='mmdet.YOLOAnchorGenerator',
-                     base_sizes=[[(10, 13), (16, 30), (33, 23)],
-                                 [(30, 61), (62, 45), (59, 119)],
-                                 [(116, 90), (156, 198), (373, 326)]],
-                     strides=[8, 16, 32]),
-                 bbox_coder: ConfigType = dict(type='YOLOv5BBoxCoder'),
-                 loss_cls: ConfigType = dict(
-                     type='mmdet.CrossEntropyLoss',
-                     use_sigmoid=True,
-                     reduction='sum',
-                     loss_weight=1.0),
-                 loss_bbox: ConfigType = dict(
-                     type='mmdet.GIoULoss', reduction='sum', loss_weight=5.0),
-                 loss_obj: ConfigType = dict(
-                     type='mmdet.CrossEntropyLoss',
-                     use_sigmoid=True,
-                     reduction='sum',
-                     loss_weight=1.0),
-                 train_cfg: OptConfigType = None,
-                 test_cfg: OptConfigType = None,
-                 init_cfg: OptMultiConfig = None):
-        super().__init__(
-            head_module=head_module,
-            prior_generator=prior_generator,
-            bbox_coder=bbox_coder,
-            loss_cls=loss_cls,
-            loss_bbox=loss_bbox,
-            loss_obj=loss_obj,
-            train_cfg=train_cfg,
-            test_cfg=test_cfg,
-            init_cfg=init_cfg)
-
-    def special_init(self):
-        """Since YOLO series algorithms will inherit from YOLOv5Head, but
-        different algorithms have special initialization process.
-
-        The special_init function is designed to deal with this situation.
-        """
-        pass
+                 *args,
+                 simota_candidate_topk: int = 10,
+                 simota_iou_weight: float = 3.0,
+                 simota_cls_weight: float = 1.0,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.assigner = BatchYOLOv7Assigner(
```

Comment:
need to build assigner by config?

Reply:
I don't think it is necessary because there is no reusability